unit_tests/fake/fake_thread_utils.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "adjustable_array.h"
+#include "critical_section_tracker.hpp"
+#include "error_context.hpp"
+#include "lockfree_transaction_system.hpp"
+#include "log_compress.h"
+#include "log_system_tran.hpp"
+#include "page_buffer.h"
+#include "resource_tracker.hpp"
+#include "system_parameter.h"
```

Comment:
Many definitions here seem to be not needed or may be moved to another file.

  - Remove all unnecessary functions
  - Move er_ functions to fake_error_manager.cpp
  - Move prm_ functions to fake_system_parameter.cpp
  - Move cubthread::entry functions to fake_thread_entry.cpp
  - Cleanup all unnecessary headers afterwards